### PR TITLE
CP-871 Utilize dart_dev for tooling, refactor server component

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Issue
Dart_dev is public and we should leverage it to make w_transport's tooling easier. Additionally, the server used for examples and integration testing is in need of refactoring for improved ease of use and ability to handle multiple types of connections (HTTP & WebSocket)

## Changes
**Source:**
- Dart formatter

**Examples:**
- Simplify the server status component since the servers were consolidated into a single server
- Update the proxy paths

**Tests**
- Split up into unit tests and integration tests

**Server**
- Consolidate http, proxy http, and (eventually) websocket server into a single server
- Get rid of `shelf` dependency
- Refactor server into a class that can be started and stopped easily so that it integrates with dart_dev well

**Tooling**
- Configure dart_dev to run analyzer, formatter, examples, tests, and coverage
- Examples, tests, and coverage all start the server prior to and stop it after completion

## Areas of Regression
- Examples
- Tooling

## Testing
- `ddev analyze` - no issues
- `ddev format` - no changes
- `ddev test --integration` - all tests pass
- `ddev coverage --integration` - coverage succeeds with ~97% coverage

## Code Review
@trentgrover-wf
@maxwellpeterson-wf 
@dustinlessard-wf
@jayudey-wf

### CODE REVIEW

**Comment by trentgrover-wf** [comment]
+1, but looks like travis CI can't clone dart_dev

**Comment by evanweible-wf** [comment]
@trentgrover-wf CI is passing

**Comment by trentgrover-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
+1

File: lib/src/http/common/util.dart [hunk 1]
```diff
@@ -8,14 +8,18 @@ StreamTransformer decodeAttempt(Encoding encoding) {
...
-        } catch (e) {}
-        controller.add(data);
-      }, onError: controller.addError, onDone: () {
-        controller.close();
-      }, cancelOnError: cancelOnError);
+      subscription = input.listen(
+          (data) {
+            try {
+              data = encoding.decode(data);
+            } catch (e) {}
```
Comment (dustinlessard-wf) [inline lib/src/http/common/util.dart]: I was a bit nervous with this error-suppression, but after seeing how this function is used - it makes sense.

**Comment by dustinlessard-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
@jayudey-wf ready for merge.

File: .travis.yml [hunk 2]
```diff
@@ -6,6 +6,8 @@ before_install:
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
 script:
-  - dart --checked test/run_tests.dart -p vm -p content-shell --verbose
-  - ./tool/analyze.sh
-  - ./tool/dartfmt.sh -e
\ No newline at end of file
+  - pub run dart_dev format --check
```
Comment (jayudey-wf) [inline .travis.yml]: will unit tests still be run in the travis script?

Comment (evanweible-wf) [inline .travis.yml]: yes, I believe travis runs all of the scripts in this section even if one or more of them fail

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - ran through commands in testing suggestions and all worked as expected
 - examples work as expected in dartium and chrome
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Comment by teresarevious-wf** [comment]
@rmconsole-wf

**Pull request merged**